src/content.ts

```diff
@@ -160,3 +163,37 @@ function showQrCode(msg: string) {
       'toolbar=no, location=no, status=no, menubar=no, scrollbars=yes, copyhistory=no, width=400, height=200, left=' +
           left + ',top=' + top);
 }
+
+function pasteCode(code: string) {
+  const _inputBoxes = document.getElementsByTagName('input');
+  const inputBoxes: HTMLInputElement[] = [];
+  for (let i = 0; i < _inputBoxes.length; i++) {
+    if (_inputBoxes[i].type === 'text' || _inputBoxes[i].type === 'number') {
+      inputBoxes.push(_inputBoxes[i]);
+    }
+  }
+  if (!inputBoxes.length) {
+    return;
+  }
+  const identities = ['2fa', 'otp', 'authenticator', 'factor'];
+  for (const inputBox of inputBoxes) {
+    for (const identity of identities) {
+      if (inputBox.name.toLowerCase().indexOf(identity) >= 0 ||
+          inputBox.id.toLowerCase().indexOf(identity) >= 0) {
+        inputBox.value = code;
+        return;
+      }
+    }
+  }
+
+  const activeInputBox = document.activeElement.tagName === 'INPUT' ?
+      document.activeElement :
+      null;
+  if (activeInputBox) {
+    (activeInputBox as HTMLInputElement).value = code;
+    return;
+  }
+  const firstInputBox = inputBoxes[0];
+  firstInputBox.value = code;
```

Comment:
This is destructive to data already in the input box. Either remove this section and let user paste code, or check if data is already in input box. This also goes for line 193